modules/dev-utils/src/main/java/org/apache/ignite/development/utils/DataEntryWrapper.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.development.utils;
+
+import org.apache.ignite.internal.pagemem.wal.record.DataEntry;
+import org.apache.ignite.internal.pagemem.wal.record.UnwrappedDataEntry;
+import org.apache.ignite.internal.util.typedef.internal.SB;
+import org.jetbrains.annotations.Nullable;
+
+import static java.lang.String.valueOf;
+import static java.util.Objects.isNull;
+import static org.apache.ignite.development.utils.ProcessSensitiveData.HASH;
+import static org.apache.ignite.development.utils.ProcessSensitiveData.MD5;
+import static org.apache.ignite.development.utils.ProcessSensitiveDataUtils.md5;
+
+/**
+ * Wrapper {@link DataEntry} for sensitive data output.
+ */
+class DataEntryWrapper extends DataEntry {
+    /** Unwrapped DataEntry. */
+    @Nullable private final UnwrappedDataEntry unwrappedDataEntry;
+
+    /** Strategy for the processing of sensitive data. */
+    private final ProcessSensitiveData sensitiveData;
+
+    /**
+     * Constructor.
+     *
+     * @param dataEntry          Instance of {@link DataEntry}.
+     * @param sensitiveData      Strategy for the processing of sensitive data.
+     */
+    public DataEntryWrapper(
+        DataEntry dataEntry,
+        ProcessSensitiveData sensitiveData
+    ) {
+        super(
+            dataEntry.cacheId(),
+            dataEntry.key(),
+            dataEntry.value(),
+            dataEntry.op(),
+            dataEntry.nearXidVersion(),
+            dataEntry.writeVersion(),
+            dataEntry.expireTime(),
+            dataEntry.partitionId(),
+            dataEntry.partitionCounter()
+        );
+
+        this.sensitiveData = sensitiveData;
+
+        this.unwrappedDataEntry = UnwrappedDataEntry.class.isInstance(dataEntry) ?
```

Comment:
Why not "instanceof"?

Reply:
There's no overhead, it's just my style.

Reply:
Well, technically there is an overhead until the moment when JIT realizes that this code can be replaced with "instanceof", but that wasn't the reason why I mentioned this code :)